github-pullrequest-plugin/src/main/java/com/github/kostyasha/github/integration/branch/events/impl/GitHubBranchCommitChecks.java

```diff
@@ -0,0 +1,141 @@
+ * GitHub to retrieve.
+ * <p>
+ * Commit data is retrieved and then passed to each implementing instance of <code>GitHubBranchCommitCheck</code> to determine information
+ * about the commit should trigger a build.
+ * </p>
+ * <p>
+ * The implementation itself should return a <code>GitHubBranchCause</code> object indicating the build should be skipped, otherwise
+ * it should return <code>null</code>.
+ * </p>
+ *
+ * @author Kanstantsin Shautsou
+ * @author Jae Gangemi
+ */
+public class GitHubBranchCommitChecks extends GitHubBranchEvent {
+    private static final String DISPLAY_NAME = "Commit Checks";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GitHubBranchCommitChecks.class);
+
+    private List<GitHubBranchCommitCheck> checks = new ArrayList<>();
+
+    /**
+     * For groovy UI
+     */
+    @Restricted(value = NoExternalUse.class)
+    public GitHubBranchCommitChecks() {
+    }
+
+    @DataBoundConstructor
+    public GitHubBranchCommitChecks(List<GitHubBranchCommitCheck> checks) {
+        this.checks = checks;
+    }
+
+    @Override
+    public GitHubBranchCause check(GitHubBranchTrigger trigger, GHBranch remoteBranch, @CheckForNull GitHubBranch localBranch,
+            GitHubBranchRepository localRepo, TaskListener listener)
+        throws IOException {
+        if (localBranch == null) {
+            LOGGER.debug("First build of branch [%s] detected.", remoteBranch.getName());
+            return null;
+        }
+
+        GHCompare.Commit[] commits = getComparedCommits(localBranch, remoteBranch);
+        List<GitHubBranchCause> causes = checks.stream()
+                .map(event -> event.check(remoteBranch, localRepo, commits))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        String name = remoteBranch.getName();
+        if (causes.isEmpty()) {
+            LOGGER.debug("Commits for branch [{}] had no effect no triggering build.", name);
+            return null;
+        }
+
+        GitHubBranchCause cause = causes.get(0);
+        LOGGER.info("Building branch [{}] skipped due to commit check: {}", name, cause.getReason());
+
+        return cause;
+    }
+
+    // visible for testing to avoid complex mocking
+    GHCompare.Commit[] getComparedCommits(GitHubBranch localBranch, GHBranch remoteBranch) throws IOException {
+        String previous = localBranch.getCommitSha();
+        String current = remoteBranch.getSHA1();
+
+        LOGGER.debug("Comparing previous hash [{}] with current hash [{}]", previous, current);
+        return remoteBranch.getOwner()
+                .getCompare(previous, current)
+                .getCommits();
+    }
+
+    @Nonnull
+    public List<GitHubBranchCommitCheck> getChecks() {
+        if (isNull(checks)) {
+            checks = new ArrayList<>();
+        }
+        return checks;
+    }
+
+    @CheckForNull
+    void setChecks(List<GitHubBranchCommitCheck> checks) {
```

Comment:
No) it makes no sense. this annotations is only for arguments or for returned value. It makes no sense here.